app/scripts/services/quota.js

```diff
@@ -260,23 +262,59 @@ angular.module("openshiftConsole")
             if (quotaKey === 'resourcequotas') {
               return false;
             }
+            if (!typesToCheck || _.includes(typesToCheck, quotaKey)) {
+              hard = usageValue(hard);
+              if (!hard) {
+                return false;
+              }
 
-            hard = usageValue(hard);
-            if (!hard) {
-              return false;
-            }
+              var used = usageValue(_.get(q, ['used', quotaKey]));
+              if (!used) {
+                return false;
+              }
 
-            var used = usageValue(_.get(q, ['used', quotaKey]));
-            if (!used) {
-              return false;
+              return hard <= used;
             }
-
-            return hard <= used;
           });
         };
         return _.some(quotas, isExceeded) || _.some(clusterQuotas, isExceeded);
     };
 
+    // Same as above but only looking at storage items: requests.storage, persistentvolumeclaims
+    //   Warn if you are at quota or over any storage quota for any resource.
+    var isAnyStorageQuotaExceeded = function(quotas, clusterQuotas) {
+      return isAnyQuotaExceeded(quotas, clusterQuotas, ['requests.storage', 'persistentvolumeclaims']);
+    };
+
+   // Check if requested quota will exceed any quotas if attempted
+    var willRequestExceedQuota = function(quotas, clusterQuotas, requestedQuotaKey, request) {
+      var isExceeded = function(quota) {
+        var q = quota.status.total || quota.status;
+        var value = usageValue(request);
+        if (!requestedQuotaKey) {
+          return false;
+        }
+        return _.some(q.hard, function(hard, quotaKey) {
```

Comment:
so you know the exact key you want to check, instead of iterating over every quotaKey with _.some can't you just directly access it?

```
var hard = _.get(q.hard, quotaKey);
hard = usageValue(hard);
// etc...
```